lib/posix/pthread.c

```diff
@@ -595,7 +595,32 @@ int pthread_setcanceltype(int type, int *oldtype)
 
 	return 0;
 }
+/**
+ * @brief Create a cancellation point in the calling thread.
+ *
+ * See IEEE 1003.1
+ */
+void pthread_testcancel(void)
```

Comment:
This function doesn't seem to account for the `cancelstate`

Reply:
Do you mean to check the cancel state if it's on PTHREAD_CANCEL_ENABLE ?

Reply:
> Do you mean to check the cancel state if it's on PTHREAD_CANCEL_ENABLE/PTHREAD_CANCEL_DISABLE ?

yes